codegen/src/main/java/com/paypal/selion/plugins/CodeGeneratorMojo.java

```diff
@@ -124,6 +125,11 @@ private String pathToFolder(File file) {
      */
     public void execute() throws MojoExecutionException {
 
+        // iff basePackage is null, empty, or just whitespace reset it to the default value
+        basePackage = StringUtils.defaultString(basePackage, "com.paypal.selion.testcomponents");
```

Comment:
@mach6 

If the user purposefully or by mistake ended up overriding the value of `basePackage` with an empty value [ Can happen in scenarios wherein the user decides to use a Maven property to define this value, but forgets to provide a value for the Maven property via a JVM argument ] and if the user intentionally had all of his/her Yaml files under `GUIData` then shouldn't we be letting the user that some did go wrong rather than us trying to fix the problem on our own with a `default` value and thus going against what the user might have been expecting ? 
In the scenario that I explained, the application behaved in a way which the user didn't expect i.e., he thought he had provided a value for the `basePackage` via a Maven property and yet he/she sees the package name beginning with `default`, they would be misguided thinking that this is a bug in SeLion whereas the actual problem resided in their own test project ? Wouldn't that be an anomaly ?